### PR TITLE
Changed the gofmt step to use golangci-lint instead for CVO

### PR DESCRIPTION
**Pull request by arjunrn**
Switch to using [golangci-lint](golangci-lint.run/) for detection of code linting and formatting issues.

### CODE REVIEW

File: ci-operator/config/openshift/cluster-version-operator/openshift-cluster-version-operator-master.yaml [hunk 1]
```diff
@@ -39,8 +39,8 @@ tag_specification:
   name: "4.9"
   namespace: ocp
 tests:
-- as: gofmt
-  commands: go fmt ./... && git diff --exit-code
+- as: golangci-lint
+  commands: export GOLANGCI_LINT_CACHE=/tmp/.cache make lint
```
Comment (wking) [inline ci-operator/config/openshift/cluster-version-operator/openshift-cluster-version-operator-master.yaml:43]: Looking at [the `lint` recipe][1], it's not clear to me whether or not it prints a patch you can apply locally to fix the issues.  If it doesn't, can we add a new job and keep the `gofmt` job?

[1]: https://github.com/openshift/cluster-version-operator/blob/e3474f2c24675fff2bd07061c95e4062395b0f9c/Makefile#L34

Comment (arjunrn) [reply ci-operator/config/openshift/cluster-version-operator/openshift-cluster-version-operator-master.yaml:47]: Done

Comment (arjunrn) [reply ci-operator/config/openshift/cluster-version-operator/openshift-cluster-version-operator-master.yaml:47]: Most of the linters cannot output a diff like `gofmt`. Only some can so I didn't enable that.

**Review by vrutkovs — APPROVED** [review]
/lgtm

**Pull request merged**